### PR TITLE
Feature/p7xfx7ib auth

### PR DESCRIPTION
**Pull request by ZrodoZrodo**

### CODE REVIEW

**Review by F-Wojciechowski — APPROVED** [review]

File: src/student/student.controller.ts [hunk 1]
```diff
@@ -18,16 +18,19 @@ import { UpdateStudentDetailsDto } from './dto/update-student-details.dto';
...
 export class StudentController {
   constructor(@Inject(StudentService) private studentService: StudentService) {}
 
+  @UseGuards(JwtAuthGuard)
   @Get('/available-students')
   async getAvailableStudents(): Promise<Omit<OneStudentResponse, 'degrees'>[]> {
     return await this.studentService.getStudents(StudentStatus.available);
   }
 
+  @UseGuards(JwtAuthGuard)
```
Comment (madradeus) [inline src/student/student.controller.ts:55]: Why do we protect  by guard only these two EPs, but rest not?

File: src/auth/auth.controller.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { AuthService } from './auth.service';
+import { JwtAuthGuard } from './jwt-auth.guard';
+
+@Controller('auth')
+export class AuthController {
+  constructor(private authService: AuthService) {}
+
+  @UseGuards(LocalAuthGuard)
+  @Post('/login')
+  async login(@Request() req) {
```
Comment (madradeus) [inline src/auth/auth.controller.ts]: Should we add return type?
If you return entire ...req.user object, is there potentially too much data like  password or currentToken ?

File: src/auth/constants.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export const jwtConstants = {
```
Comment (madradeus) [inline src/auth/constants.ts]: in my opinion this should not be in the repository. It is sensitive data, so I would put in in config file

File: src/student/student.controller.ts (hunk 1 above)
Comment (ZrodoZrodo) [reply src/student/student.controller.ts:55]: this is only for tests now we must choose witch endpoints we must protect

**Review by madradeus — APPROVED** [review]

**Pull request merged**